src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test of dynamic stub import API."""
+
+import contextlib
+import functools
+import logging
+import multiprocessing
+import os
+import sys
+import unittest
+
+
+@contextlib.contextmanager
+def _grpc_tools_unimportable():
+    original_sys_path = sys.path
+    sys.path = [path for path in sys.path if "grpcio_tools" not in path]
+    try:
+        import grpc_tools
+    except ImportError:
+        pass
+    else:
+        del grpc_tools
+        sys.path = original_sys_path
+        raise unittest.SkipTest("Failed to make grpc_tools unimportable.")
```

Comment:
Curious: what could be causing this corner case?

Reply:
This is possible when `grpc_tools` is importable from a path that doesn't include the path `grpc_tools`. As far as I can tell, this should include all of our intentional inclusions on `sys.path`, but some platforms like Windows turned out to break that assumption somehow.